src/main/java/org/prebid/server/bidder/bigoad/BigoadBidder.java

```diff
@@ -0,0 +1,150 @@
+
+    private static final TypeReference<ExtPrebid<?, ExtImpBigoad>> BIGOAD_EXT_TYPE_REFERENCE =
+            new TypeReference<>() {
+            };
+    private static final String SSP_ID_MACRO = "{{SspId}}";
+    private static final String OPEN_RTB_VERSION = "2.5";
+
+    private final String endpointUrl;
+    private final JacksonMapper mapper;
+
+    public BigoadBidder(String endpointUrl, JacksonMapper mapper) {
+        this.endpointUrl = HttpUtil.validateUrl(Objects.requireNonNull(endpointUrl));
+        this.mapper = Objects.requireNonNull(mapper);
+    }
+
+    @Override
+    public Result<List<HttpRequest<BidRequest>>> makeHttpRequests(BidRequest request) {
+        final List<Imp> modifiedImps = new ArrayList<>(request.getImp());
+
+        final Imp firstImp = modifiedImps.getFirst();
+        final ExtImpBigoad extImpBigoad;
+        try {
+            extImpBigoad = parseImpExt(firstImp);
+        } catch (PreBidException e) {
+            return Result.withError(BidderError.badInput(e.getMessage()));
+        }
+
+        modifiedImps.set(0, modifyImp(firstImp, extImpBigoad));
+        return Result.withValue(BidderUtil.defaultRequest(
+                updateBidRequest(request, modifiedImps),
+                headers(),
+                makeEndpointUrl(extImpBigoad),
+                mapper));
+    }
+
+    private ExtImpBigoad parseImpExt(Imp imp) throws PreBidException {
+        try {
+            return mapper.mapper().convertValue(imp.getExt(), BIGOAD_EXT_TYPE_REFERENCE).getBidder();
+        } catch (IllegalArgumentException e) {
+            throw new PreBidException("imp %s: unable to unmarshal ext.bidder: %s"
+                    .formatted(imp.getId(), e.getMessage()));
+        }
+    }
+
+    private Imp modifyImp(Imp imp, ExtImpBigoad extImpBigoad) {
+        return imp.toBuilder().ext(mapper.mapper().valueToTree(extImpBigoad)).build();
+    }
+
+    private static BidRequest updateBidRequest(BidRequest bidRequest, List<Imp> imps) {
+        return bidRequest.toBuilder().imp(imps).build();
+    }
+
+    private static MultiMap headers() {
+        return HttpUtil.headers().add(HttpUtil.X_OPENRTB_VERSION_HEADER, OPEN_RTB_VERSION);
+    }
+
+    private String makeEndpointUrl(ExtImpBigoad extImpBigoadx) {
+        final String safeSspId = HttpUtil.encodeUrl(StringUtils.trimToEmpty(extImpBigoadx.getSspId()));
+        return endpointUrl.replace(SSP_ID_MACRO, safeSspId);
+    }
+
+    @Override
+    public Result<List<BidderBid>> makeBids(BidderCall<BidRequest> httpCall, BidRequest bidRequest) {
+        try {
+            final List<BidderError> errors = new ArrayList<>();
+            final BidResponse bidResponse = parseBidResponse(httpCall.getResponse());
+            return Result.of(extractBids(bidResponse, errors), errors);
+        } catch (PreBidException e) {
+            return Result.withError(BidderError.badServerResponse(e.getMessage()));
+        }
```

Comment:
can we just catch both PreBidException and DecodeException here and not make a separate `parseBidResponse` method?